File: T2/EB_BDD_GR9/03.RESTFULL_JAVA_EUREKABANK_GR09/03.RESTFULL_JAVA_EUREKABANK_GR09/02.CLICON/BDD_RESTFUL_JAVA_CLICON_G09/src/main/java/ec/edu/monster/vista/MovimientoView.java
/*
 * Vista para mostrar movimientos bancarios
 * Diseño mejorado similar al cliente .NET
 */
package ec.edu.monster.vista;

import ec.edu.monster.modelo.Movimiento;
import java.text.DecimalFormat;
import java.text.SimpleDateFormat;
import java.util.List;

/**
 * Clase para mostrar movimientos en formato de tabla
 * @author Grupo Monster G09
 */
public class MovimientoView {
    
    private static final DecimalFormat formatoMoneda = new DecimalFormat("#,##0.00");
    private static final SimpleDateFormat formatoFecha = new SimpleDateFormat("dd/MM/yyyy");
    
    // Colores (usando los mismos del cliente)
    public static final String COLOR_1 = "\033[38;2;58;180;217m"; // Azul claro
    public static final String COLOR_3 = "\033[38;2;246;222;136m"; // Amarillo claro
    public static final String BLANCO = "\033[97m";
    public static final String RESET = "\033[0m";
    
    /**
     * Obtiene la descripción del tipo de movimiento basado en el código
     */
    private static String obtenerDescripcionTipo(String tipocodigo) {
        if (tipocodigo == null) return "";
        switch (tipocodigo) {
            case "001": return "Apertura de Cuenta";
            case "002": return "Cancelar Cuenta";
            case "003": return "Deposito";
            case "004": return "Retiro";
            case "005": return "Interes";
            case "006": return "Mantenimiento";
            case "007": return "ITF";
            case "008": return "Transferencia";
            case "009": return "Transferencia";
            case "010": return "Cargo por Movimiento";
            default: return tipocodigo;
        }
    }
    
    /**
     * Muestra los movimientos en formato de tabla
     * @param movimientos Lista de movimientos a mostrar
     * @param anchoCuadro Ancho del cuadro para centrar
     */
    public static void mostrarMovimientos(List<Movimiento> movimientos, int anchoCuadro) {
        if (movimientos == null || movimientos.isEmpty()) {
            return;
        }
        
        int width = 120; // Ancho estándar de consola
        int startPos = (width - anchoCuadro) / 2;
        if (startPos < 0) startPos = 0;
        
        // Asignación de anchos para columnas
        int anchoCuenta = 15;
        int anchoNroMov = 8;
        int anchoFecha = 12;
        int anchoTipo = 20;
        int anchoAccion = 10;
        int anchoImporte = anchoCuadro - (anchoCuenta + anchoNroMov + anchoFecha + anchoTipo + anchoAccion + 6);
        
        // Línea superior
        System.out.print(" ".repeat(startPos));
        System.out.println(COLOR_1 + "=".repeat(anchoCuadro) + RESET);
        
        // Encabezado
        System.out.print(" ".repeat(startPos));
        String encabezado = String.format(
            "%-" + anchoCuenta + "s %-" + anchoNroMov + "s %-" + anchoFecha + "s %-" + anchoTipo + "s %-" + anchoAccion + "s %" + anchoImporte + "s",
            "Cuenta", "Nro Mov.", "Fecha", "Tipo", "Acción", "Importe"
        );
        System.out.println(COLOR_1 + encabezado + RESET);
        
        // Línea separadora
        System.out.print(" ".repeat(startPos));
        System.out.println(COLOR_1 + "=".repeat(anchoCuadro) + RESET);
        
        // Mostrar cada movimiento
        for (Movimiento mov : movimientos) {
            String tipoDescripcion = obtenerDescripcionTipo(mov.getTipocodigo());
            String tipo = tipoDescripcion.length() > anchoTipo 
                ? tipoDescripcion.substring(0, anchoTipo) 
                : tipoDescripcion;
            
            String accion = mov.getAccion() != null && mov.getAccion().length() > anchoAccion 
                ? mov.getAccion().substring(0, anchoAccion) 
                : (mov.getAccion() != null ? mov.getAccion() : "");
            
            String fechaStr = mov.getMovifecha() != null 
                ? formatoFecha.format(mov.getMovifecha()) 
                : "N/A";
            
            String importeStr = "$" + formatoMoneda.format(mov.getMoviimporte());
            
            System.out.print(" ".repeat(startPos));
            String linea = String.format(
                "%-" + anchoCuenta + "s %-" + anchoNroMov + "d %-" + anchoFecha + "s %-" + anchoTipo + "s %-" + anchoAccion + "s %" + anchoImporte + "s",
                mov.getCuencodigo() != null ? mov.getCuencodigo() : "",
                mov.getMovinumero(),
                fechaStr,
                tipo,
                accion,
                importeStr
            );
            System.out.println(BLANCO + linea + RESET);
        }
        
        // Línea inferior
        System.out.print(" ".repeat(startPos));
        System.out.println(COLOR_1 + "=".repeat(anchoCuadro) + RESET);
    }
}

File: T2/EB_BDD_GR9/02.SOAP_DOTNET_EUREKABANK_GR09/02.SOAP_DOTNET_EUREKABANK_GR09/02.CLICON/BDD_SOAP_DOTNET_CLICON_G09/BDD_SOAP_DOTNET_CLICON_G09/ec/edu/monster/vista/MovimientoView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using BDD_SOAP_DOTNET_CLICON_G09.ec.edu.monster.modelo;

namespace BDD_SOAP_DOTNET_CLICON_G09.ec.edu.monster.vista
{
    public class MovimientoView
    {
        public static void MostrarMovimientos(List<CliCon_Movimiento> movimientos, int anchoCuadro)
        {
            if (movimientos == null || movimientos.Count == 0)
            {
                return;
            }

            int width = Console.WindowWidth;
            int startPos = (width - anchoCuadro) / 2;

            Console.OutputEncoding = System.Text.Encoding.UTF8;
            Console.ForegroundColor = ConsoleColor.Blue;

            // Línea superior
            Console.SetCursorPosition(startPos, Console.CursorTop);
            Console.WriteLine(new string('=', anchoCuadro));

            // Asignación de anchos para columnas
            int anchoCuenta = 15;
            int anchoNroMov = 8;
            int anchoFecha = 12;
            int anchoTipo = 20;
            int anchoAccion = 10;
            int anchoImporte = anchoCuadro - (anchoCuenta + anchoNroMov + anchoFecha + anchoTipo + anchoAccion + 6);

            // Encabezado
            Console.SetCursorPosition(startPos, Console.CursorTop);
            string encabezado = string.Format(
                "{0,-" + anchoCuenta + "} {1,-" + anchoNroMov + "} {2,-" + anchoFecha + "} {3,-" + anchoTipo + "} {4,-" + anchoAccion + "} {5," + anchoImporte + "}",
                "Cuenta", "Nro Mov.", "Fecha", "Tipo", "Acción", "Importe");
            Console.WriteLine(encabezado);

            // Línea separadora
            Console.SetCursorPosition(startPos, Console.CursorTop);
            Console.WriteLine(new string('=', anchoCuadro));

            Console.ForegroundColor = ConsoleColor.White;

            // Mostrar cada movimiento
            foreach (var mov in movimientos)
            {
                string tipo = mov.Tipo != null && mov.Tipo.Length > anchoTipo ? mov.Tipo.Substring(0, anchoTipo) : (mov.Tipo ?? "");
                string accion = mov.Accion != null && mov.Accion.Length > anchoAccion ? mov.Accion.Substring(0, anchoAccion) : (mov.Accion ?? "");

                Console.SetCursorPosition(startPos, Console.CursorTop);
                string linea = string.Format(
                    "{0,-" + anchoCuenta + "} {1,-" + anchoNroMov + "} {2,-" + anchoFecha + "} {3,-" + anchoTipo + "} {4,-" + anchoAccion + "} {5," + anchoImporte + "}",
                    mov.Cuenta ?? "",
                    mov.NroMov,
                    mov.Fecha.ToString("dd/MM/yyyy"),
                    tipo,
                    accion,
                    mov.Importe.ToString("N2")
                );
                Console.WriteLine(linea);
            }

            // Línea inferior
            Console.ForegroundColor = ConsoleColor.Blue;
            Console.SetCursorPosition(startPos, Console.CursorTop);
            Console.WriteLine(new string('=', anchoCuadro));

            Console.ResetColor();
        }

        public void MostrarMensaje(string mensaje)
        {
            Console.WriteLine(mensaje);
        }
    }
}

